src/connectors/impls/tcp/client.rs

```diff
@@ -215,18 +223,14 @@ impl Sink for TcpClientSink {
         let buf_size = self.config.buf_size;
 
         // connect TCP stream
-        let stream = TcpStream::connect((
-            self.config.url.host_or_local(),
-            self.config.url.port_or_dflt(),
-        ))
-        .await?;
+        let stream = tcp_client_socket(&self.config.url, &self.config.socket_options).await?;
         let local_addr = stream.local_addr()?;
+        let peer_addr = stream.peer_addr()?;
         // this is known to fail on macOS for IPv6.
         // See: https://github.com/rust-lang/rust/issues/95541
         //if let Some(ttl) = self.config.ttl {
```

Comment:
not part of this but we could guard this for mac os instead of completely ignoring TTL

Reply:
lets do this in another PR then

Reply:
makes sense